### DISCUSSION TITLE
out of license

### DISCUSSION THREAD

**Discussion by SirDank** [dashingsoft/pyarmor#1314] [Q&A]
```
INFO     Python 3.11.3
INFO     Pyarmor 8.2.5 (trial), 000000, non-profits
INFO     Platform windows.x86_64
DEBUG    native platform windows.amd64
DEBUG    home path: C:\Users\SirDank\.pyarmor
DEBUG    command options: {'no_runtime': False, 'inputs': ['nqueens.py']}
DEBUG    install plugin: CodesignPlugin
INFO     search inputs ...
INFO     find script nqueens.py
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'windows.amd64'}
INFO     write dist\pyarmor_runtime_000000\pyarmor_runtime.pyd
INFO     patch runtime file
DEBUG    patching runtime data at 511008
DEBUG    patch runtime file OK
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "nqueens"
INFO     obfuscating file nqueens.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    patch co
DEBUG    unknown error, please check pyarmor.error.log
ERROR    out of license
```

**Comment by jacob-hansen** [comment]
I am also running into issues using my license. Not sure if it relates. I purchased a group license, I can successfully register the devices. After a few days of use, it disassociates my “pyarmor-device-####.#.zip”. Then when I go to use it again, I get an error “ERROR this regale is not for this device”. 
I’m on windows 11, fully updated pyarmor.

**Comment by jondy** [comment]
@jacob-hansen please register this device again. If the hardware information changed, please register this device again. When upgrading  pyarmor, it may need register this device.

**Reply by jacob-hansen** [reply]
Hi Jondy,

Thank you for the help! I will be registering the device again, but every time it generates the same device file (probably the MAC address?). Thus, like the last time I tried, I don’t think just generating another license will be a long term solution. I’ll quickly go through all 100 devices in my license. I assume this was not intentional? 

-Jacob

**Reply by jacob-hansen** [reply]
Maybe there was an issue that when I move the files, it no longer registers as the same device. I've registered the device again, and I'll see if it continues to work if I don't move the files.

**Reply by jondy** [reply]
The generated file should work if device name is same and there is no changed hardware in the device.

**Comment by SirDank** [comment]
@jondy could you help me with this? what exactly do I need to do?

**Comment by jondy** [comment]
@SirDank Please search `out of license` in the online documentation.
